### PR TITLE
Adjust Heading Size and Spacing

### LINKED ISSUE #159: Adjust MUI's heading sizes to match the design
As discussed internally, let's:

* Adjust MUI's heading sizes so we can use `h1-6` with the current design.
* Change the `h1` in the `Layout` component to use `h1` variant.
* Change the `h3` in the `LabelValuePanel` to use `h2`? (unless we expect other higher-level headings on the page)

### PR DESCRIPTION
**Pull request by dhurlburtusa**
Adjust Heading Size and Spacing.

Resolves #159.

## How has this been tested?

Manually, visually tested.

### CODE REVIEW

File: components/Headings.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Typography from '@material-ui/core/Typography'
```
Comment (dhurlburtusa) [inline components/Headings.stories.tsx:1]: Story added so we can see all headings together.

File: components/Layout.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ const useStyles = makeStyles((theme: Theme) =>
       flex: '1 0',
     },
     contentTitle: {
-      margin: theme.spacing(4, 0, 2, 0),
+      margin: theme.spacing(3, 0, 2, 0),
```
Comment (dhurlburtusa) [inline components/Layout.tsx:61]: The top margin looked excessive. Reduced a bit.

File: styles/theme.ts [hunk 3]
```diff
@@ -21,6 +21,11 @@ declare module '@material-ui/core/styles' {
   }
 }
 
+const headingDefaults = {
```
Comment (dhurlburtusa) [inline styles/theme.ts:24]: These defaults are borrowed from Bootstrap 4.

File: styles/theme.ts [hunk 4]
```diff
@@ -75,6 +81,32 @@ const theme = createMuiTheme({
       main: '#194661',
     },
   },
+  typography: {
```
Comment (dhurlburtusa) [inline styles/theme.ts:84]: These font-sizes are borrowed from Bootstrap 4.

File: components/Layout.tsx (hunk 2 above)
Comment (jessie-ross) [reply components/Layout.tsx:61]: It is subjective but I felt that was a good amount of top margin, it gave a nice vertical rhythm.

Comment (yanirs) [reply components/Layout.tsx:61]: Yeah, this is also unrelated to #159 so let's keep it out of this PR.

```suggestion
      margin: theme.spacing(4, 0, 2, 0),
```

File: styles/theme.ts (hunk 4 above)
Comment (yanirs) [reply styles/theme.ts:87]: The sizes still feel a bit excessive, but we can iterate.

**Pull request merged**